src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -997,6 +1000,143 @@ public static IEnumerable<object[]> VerifyUploadServersStreamsAndExpectedData
             }
         }
 
+        [Fact]
+        public async Task PostAsync_ResponseContentRead_RequestContentDisposedAfterResponseBuffered()
```

Comment:
That's what https://github.com/dotnet/corefx/pull/9024/files#diff-06a6483ff6bb478e06f199d69522bcd4R1050 is for.